### PR TITLE
Improved CI

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Add coverage, coveralls, and other badges.

:)

### CODE REVIEW

**Comment by AdrieanKhisbe** [comment]
(I hesitated updating the depenancies, but this causes the following npm warnings:

```
npm WARN unmet dependency /Users/abecchis/git/-tmp-fork/chairo/node_modules/lab requires items@'1.x.x' but will load
npm WARN unmet dependency /Users/abecchis/git/-tmp-fork/chairo/node_modules/items,
npm WARN unmet dependency which is version 2.0.0
npm WARN unmet dependency /Users/abecchis/git/-tmp-fork/chairo/node_modules/seneca/node_modules/seneca-transport/node_modules/wreck requires hoek@'2.x.x' but will load
npm WARN unmet dependency /Users/abecchis/git/-tmp-fork/chairo/node_modules/hoek,
npm WARN unmet dependency which is version 3.0.4
npm WARN unmet dependency /Users/abecchis/git/-tmp-fork/chairo/node_modules/seneca/node_modules/seneca-transport/node_modules/wreck requires boom@'2.x.x' but will load
npm WARN unmet dependency /Users/abecchis/git/-tmp-fork/chairo/node_modules/boom,
npm WARN unmet dependency which is version 3.0.0
```

**Comment by Marsup** [comment]
We generally don't want coveralls in hapi projects, all projects have 100% coverage otherwise tests won't pass, that's not necessary.

File: .travis.yml [hunk 1]
```diff
@@ -3,3 +3,10 @@ language: node_js
 node_js:
   - 4
   - 5
+
+after_script:
+  - npm run cover
+
+cache:
```
Comment (Marsup) [inline .travis.yml]: Why do you insist on caching that ?

Comment (AdrieanKhisbe) [inline .travis.yml]: I'm not really insisting.

However, there are two main advantages: 
- faster build time: (you can check yourself on https://travis-ci.org/senecajs/seneca/builds, where it was set up few days ago)
- network, cpu, hence energy savings

**Comment by AdrieanKhisbe** [comment]
Indeed all hapi plugins have a 100% coverage.
However, this sign of high quality testing is not visible on the README for users.
Badges are getting more and more important for a lot of person to quickly access the quality of a project, before to dig deeper.

File: .travis.yml (hunk 1 above)
Comment (Marsup) [inline .travis.yml]: That's supposing npm does a good job at updating cached dependencies, it doesn't always.

Comment (AdrieanKhisbe) [inline .travis.yml]: It's travis taking care to do a cache, not redowlloading again the npm dependancies if `package.json` is not updated.

Comment (Marsup) [inline .travis.yml]: Yes, and you'd be using outdated dependencies.

Comment (geek) [inline .travis.yml]: Please go ahead and remove the cache part... I do like the coveralls integration... but as Marsup says, its not necessary.  Instead, can you link to a badge that says 100% coverage!

Comment (AdrieanKhisbe) [inline .travis.yml]: Haha. Wolkswagen style =)
Not necessary but usefull.

**Comment by AdrieanKhisbe** [comment]
I just removed the cache part.

Pasting below the patch with the package version update.. What you think about

```
diff --git a/package.json b/package.json
index 33f334b..ac3e9e2 100644
--- a/package.json
+++ b/package.json
@@ -14,13 +14,13 @@
     "node": ">=4.0.0"
   },
   "dependencies": {
-    "boom": "2.x.x",
-    "hoek": "2.x.x",
-    "items": "1.x.x",
-    "joi": "6.x.x",
+    "boom": "3.x.x",
+    "hoek": "3.x.x",
+    "items": "2.x.x",
+    "joi": "7.x.x",
     "jsonic": "0.2.x",
     "seneca": "0.x.x",
-    "vision": "3.x.x"
+    "vision": "4.x.x"
   },
```

**Pull request merged**